sources/api/storewolf/src/main.rs

```diff
@@ -385,9 +398,25 @@ fn populate_default_datastore<P: AsRef<Path>>(
     Ok(())
 }
 
+/// Creates a symlink given a source path and a destination path. Creates the
+/// necessary parent directories along the destination path.
+fn create_inventory_symlink<P: AsRef<Path>>(source: P, destination: P) -> Result<()> {
```

Comment:
Nit: Probably not a big deal in this particular case, but `source` and `destination` are both `P: Asref<Path>`.  This means they both must be the same underlying type; you couldn't pass a `String` and a `&PathBuf` for example.

To be a bit more flexible, you can use the same trait bound, but split into multiple params, which allows the underlying types to be different. (Excluse the formatting)
```
fn create_inventory_symlink<P1, P2>(source: P1, destination: P2) -> Result<()>
    where P1: AsRef<Path>, P2: AsRef<Path>
{
...
}
```

Reply:
Huh interesting! I was not aware that calling into the params type would assume the same underlying type _for both_. I'll make that change 👍🏼